web/utils.js

```diff
@@ -439,3 +439,173 @@ function checkEnter(event) {
         $("#sbox").submit();
     }
 }
+
+// Intelligence Window code starts from here
+document.onmousemove = function(event) {
+    if (typeof event !== 'undefined') {
+        document.intelliWindowMouseX = event.clientX;
+        document.intelliWindowMouseY = event.clientY;
+    }
+};
+
+$(document).keypress(function(e) {
+    if (document.activeElement.id === 'search' ||
+        typeof document.intelliWindow === 'undefined') {
+        return true;
+    }
+
+    if (e.which === 49) { // '1' pressed
+        if (document.intelliWindow.className === "intelli_window_style") {
+            hideIntelliWindow();
+        } else if (document.intelliWindow.className === "intelli_window_style_hide") {
+            showIntelliWindow();
+        }
+    }
+    if (e.which === 50) { // '2' pressed
+        var symbol = document.intelliWindow.symbol;
+        var highlighted_symbols_with_same_name = $("a").filter(function(index) {
+            return $(this).text() === symbol &&
+                $(this).css("background-color") === "gold";
+        })
+        if (highlighted_symbols_with_same_name.length === 0) {
+            highlightSymbol(symbol);
+        } else {
+            unhighlightSymbol(symbol);
+        }
+    }
+    return true;
+});
+
+function onMouseOverSymbol(symbol, symbolType) {
+    updateIntelliWindow(symbol, symbolType);
```

Comment:
Just another idea how to automatically use symbol highlight - if user came to file view from search page - searched words could be highlighted immediately.
At least for basic search types (regex,proximity would be too complex.)

This would need adding words to href links from search page. Classes around search context/summarizer should be able to provide list of words from search query.

--
And thanks again for this really nice feature!
Please don't take this as pushing feature requests to you, I'd just like to share ideas. And hope that somebody coming around has power and time to make it reality :)